inloop/solutions/prettyprint/tools.py

```diff
@@ -0,0 +1,303 @@
+"""
+Tools and infrastructure to convert XML to template-friendly dicts.
+
+Numerous tools such as Ant, Maven and Gradle produce test reports
+according to this format, which is described as XML DTD in the
+Javadoc for the class
+
+    org.apache.tools.ant.taskdefs.optional.junit.XMLConstants
+
+which is part of Ant (see https://github.com/apache/ant).
+"""
+from defusedxml import ElementTree as ET
+
+
+class XMLContextParser(object):
+    """
+    Manages the context parsing between Docker XML files and templates.
+
+    When created, a solution is parsed, which contains the xml we need to
+    transform and parse our data in directly processable dicts.
+    """
+
+    def __init__(self, *, solution):
+        """
+        Initializes this object. Unwraps the testoutput_set of the most recent test result
+        to the internal variable self.testoutput.set.
```

Comment:
This comment is redundant: it repeats **how** the code achieves its goal. Instead, it should describe **what** this constructors does. For example: "Create a parser for the most recent test result of the given solution."

Reply:
I fixed this in #db0111c